### PR TITLE
Reading metadata information to google buckets

### LINKED ISSUE #9075: Reading metadata information to google buckets
@rdstern @jkmusyoka I know that Chris Clarke wants to get metadata information for the API (lat, long, district, elevation). Am I right in thinking that this comes through in a data frame when you download the data from climsoft into R-Instat?

Currently, we aren't reading in any of this metadata for our stations. I was thinking we have another sub dialog on our dialog to update metadata information. What do you think?

I will make the necessary R code changes if you are happy for me to. (Although I assume this isn't a high priority for Zambia @ChrisMarsh82)

### PR DESCRIPTION
**Pull request by Vitalis95**
Fixes #9075 
@lilyclements @jkmusyoka , have a look at it

### CODE REVIEW

**Comment by jkmusyoka** [comment]
@Vitalis95 thanks for the first go at this. I have two comments
1. The name of the dialog is misleading. It should be "Export to Google Buckets" not "Export Climatic Definitions"
2. I would like to suggest a redesign of the dialog so that it is consistent with other dialogs with a similar structure. This suggestion will make it possible to update the metadata without "being forced" to export summaries as well.  For the revised dialog, main dialog should have two tabs at the top namely `Update Metadata` and `Upload Summaries`. The `Token` receiver can follow below these tabs and should be given once for both tabs. What follows after the token receiver will depend on the selected tab at the top of the dialog. See the new `import from climsoft` dialog below whose design is consistent with what I am suggesting.  It might help to understand this point.
image
image

**Comment by rdstern** [comment]
@Vitalis95 just to follow the same as in Patrick's Import from Climsoft.
The Token should be in both, but it is linked, so the user can input it in either, and it is then registered in both.  I hope that's clear?

**Comment by jkmusyoka** [comment]
@Vitalis95 the `update metadata` dialog also needs a `country` receiver. This is currently missing and so it is not possible to specify which folder in the google buckets we are updating with this info. Each country has its own folder.

**Comment by Vitalis95** [comment]
@jkmusyoka , I have added the `country` argument

**Comment by Vitalis95** [comment]
@jkmusyoka , over to you. I have tested from my end its fine

**Comment by jkmusyoka** [comment]
@Vitalis95 the `update metadata` part of the dialog works fine now. 

However the other part `Upload summaries` has regressed and returns the following error. Could you look into this? 
image

**Comment by Vitalis95** [comment]
@jkmusyoka , on the `Summaries`, you have checked both `Annual Rain `and `Annual Temperature`, so when you check `Include Summary Data `checkbox then  on the sub dialog you should also have for Annual Rain and Annual Temperature . You only have for Annual Rain.

**Comment by jkmusyoka** [comment]
@Vitalis95 well spotted. That was my fault. 

The dialog is all done now, thanks for the hard work.

I ll do some final testing tomorrow and if all goes well, I ll approve so the changes are merged. 

Thanks

**Review by jkmusyoka — APPROVED** [review]
@N-thony this is now ready for merging with master. Over to you.

Thanks @Vitalis95 and @lilyclements

File: instat/sdgMetadataGoogleBuckets.vb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        ucrSelectorMetadata.SetRCode(clsUpdateMetadataInfoFunction, bReset, bCloneIfNeeded:=True)
+    End Sub
+
+    Private Sub ucrReceiverLongititude_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrReceiverLongititude.ControlValueChanged, ucrReceiverStation.ControlValueChanged, ucrReceiverLatitude.ControlValueChanged, ucrReceiverElavation.ControlValueChanged, ucrReceiverDistrict.ControlValueChanged, ucrSelectorMetadata.ControlValueChanged
+        If Not (ucrReceiverElavation.IsEmpty OrElse ucrReceiverDistrict.IsEmpty OrElse ucrReceiverLatitude.IsEmpty OrElse ucrReceiverLongititude.IsEmpty OrElse ucrReceiverStation.IsEmpty) Then
+            clsRsyntax.AddToBeforeCodes(clsUpdateMetadataInfoFunction, 1)
+        Else
+            clsRsyntax.RemoveFromBeforeCodes(clsUpdateMetadataInfoFunction)
+        End If
+    End Sub
```
[the remark this answers is not in the document]
Comment (Vitalis95) [reply instat/sdgMetadataGoogleBuckets.vb:81]: @N-thony , we don't call this sub dialog any more in the main dialog. We have made the changes on the main dialog. Is there way I can remove the sub-dialog changes from this PR?

**Pull request closed without merging**